File: app/lib/model/chat_message.dart
class ChatMessage {
  const ChatMessage({
    required this.text,
    required this.isMine,
    required this.time,
  });

  final String text;

  /// 내가 보낸 메시지면 true (오른쪽 파란색), 상대가 보낸 메시지면 false (왼쪽 회색).
  final bool isMine;

  /// 보낸 시간. (예: '오후 2:30')
  final String time;
}

File: server/src/main/kotlin/com/blueoauld/server/chat/entity/ChatMessage.kt
package com.blueoauld.server.chat.entity

import com.blueoauld.server.chat.entity.type.MessageType
import jakarta.persistence.*
import java.time.Instant

@Entity
@Table(
    name = "chat_message",
    indexes = [
        Index(name = "idx_chat_message_room_id", columnList = "room_id, id"),
    ],
)
class ChatMessage(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,

    @Column(name = "room_id", nullable = false)
    val roomId: Long,

    @Column(name = "sender_id", nullable = false)
    val senderId: Long,

    @Enumerated(EnumType.STRING)
    @Column(name = "type", nullable = false)
    val type: MessageType,

    @Column(name = "content", length = 1000, nullable = false)
    var content: String,

    @Column(name = "created_at", nullable = false, updatable = false)
    var createdAt: Instant = Instant.now(),
) {

    @PrePersist
    fun onCreate() {
        this.createdAt = Instant.now()
    }
}
